### PR TITLE
Add unity block size option

### PR DESCRIPTION
**Pull request by jpakkane**
The first commit refactors the integer option class to make this work.

### CODE REVIEW

File: docs/markdown/Builtin-options.md [hunk 1]
```diff
@@ -82,6 +82,7 @@ Using the option as-is with no prefix affects all machines. For example:
 | stdsplit                             | true          | Split stdout and stderr in test logs                           | no             |
 | strip                                | false         | Strip targets on install                                       | no             |
 | unity {on, off, subprojects}         | off           | Unity build                                                    | no             |
+| unity_size {>=2}                     | 4             | Unity file block size                                          | no             |
```
Comment (mensinda) [inline docs/markdown/Builtin-options.md]: Shouldn't this be `unity_size {>=1} `?

Comment (jpakkane) [reply docs/markdown/Builtin-options.md]: Unity size of 1 is equivalent to turning unity off altogether.

Comment (mensinda) [reply docs/markdown/Builtin-options.md]: Sure, but then the `(1, None, 4)` tuple in coredata should also be updated to reflect the documentation.

Comment (jpakkane) [reply docs/markdown/Builtin-options.md]: D'oh! Fixed.

**Comment by jpakkane** [comment]
A simple way to get the old behavior is to set the size to 10000 or so. Probably something like 100 would be sufficient, the speedups after that are probably not noticeable (note: speculation, not based on measurements).

**Comment by mensinda** [comment]
Sure this works, but it feels like a hack and people will complain that there is no proper way to enable real unity builds.

We should at least document that setting a huge number is the intended way for enabling single file unity builds if we don't directly support it. Otherwise, people will open bug reports.

**Comment by jpakkane** [comment]
Docs updated.

**Pull request merged**